Main merge release/25.12 (unblock forward merge)

alexbarghi-nv opened the pull request:
Unblocks the forward merge.

jnke2016 committed:
Update API to leverage cuVS Spectral clustering (#5326)

This PR integrates cuVS spectral clustering from upstream into cuGraph while maintaining full backward compatibility. The integration deprecates the legacy balanced cut clustering C++ and Python APIs in favor of spectral_modularity_maximization. In fact, cuVS consolidated the spectral clustering API to assign vertices to clusters that by default maximizes the modularity score

**Static Linking with Conda Fallback:**
- cuGraph is configured to statically link libcuvs for optimal performance
- The build system first attempts to use conda-installed libcuvs if available
- If libcuvs is not present in the conda environment or static linking is required, the build system automatically clones the main cuVS repository (rapidsai/cuvs) and builds it from source
- Uses the standard RAPIDS release branch (${rapids-cmake-checkout-tag}) ensuring version compatibility

Note: The header files `raft/spectral/partition.cuh` and `raft/spectral/modularity_maximization.cuh` are still necessary in `spectraly_clustering.cu` because APIs (like `raft::spectral::analyzePartition`, `raft::spectral::analyzeModularity`) are still being called in cuGraph and are not supported in cuVS. This is not an issue because those APIs are not deprecated in raft in contrary to `kmeans` and `solvers`

Authors:
  - Joseph Nke (https://github.com/jnke2016)

Approvers:
  - Anupam (https://github.com/aamijar)
  - Chuck Hastings (https://github.com/ChuckHastings)
  - Rick Ratzel (https://github.com/rlratzel)
  - Robert Maynard (https://github.com/robertmaynard)
  - Bradley Dice (https://github.com/bdice)

URL: https://github.com/rapidsai/cugraph/pull/5326

trxcllnt committed:
Use `sccache-dist` build cluster for conda and wheel builds (#5332)

RAPIDS has deployed an autoscaling cloud build cluster that can be used to accelerate building large RAPIDS projects. 

This PR updates the conda and wheel builds to use the build cluster.

This contributes to https://github.com/rapidsai/build-planning/issues/228.

Authors:
  - Paul Taylor (https://github.com/trxcllnt)

Approvers:
  - Jake Awe (https://github.com/AyodeAwe)
  - Bradley Dice (https://github.com/bdice)

URL: https://github.com/rapidsai/cugraph/pull/5332

alexbarghi-nv committed:
[FEA] Update pylibcugraph to support different temporal comparisons (#5345)

Updates pylibcugraph to support different temporal comparisons (i.e. >=, <, <=, >, last).

Authors:
  - Alex Barghi (https://github.com/alexbarghi-nv)

Approvers:
  - Joseph Nke (https://github.com/jnke2016)
  - Chuck Hastings (https://github.com/ChuckHastings)
  - Rick Ratzel (https://github.com/rlratzel)

URL: https://github.com/rapidsai/cugraph/pull/5345

alexbarghi-nv committed:
Deprecate Distributed Sampling API (#5336)

The distributed sampling API has been migrated to cuGraph-PyG, so we are deprecating it here.

Authors:
  - Alex Barghi (https://github.com/alexbarghi-nv)
  - Chuck Hastings (https://github.com/ChuckHastings)

Approvers:
  - Brad Rees (https://github.com/BradReesWork)

URL: https://github.com/rapidsai/cugraph/pull/5336

alexbarghi-nv committed:
resolve merge conflicts blocking forward merge

alexbarghi-nv committed:
fix libcuvs dependency

gforsyth commented on dependencies.yaml:
```diff
@@ -508,6 +509,11 @@ dependencies:
+    common:
+      - output_types: conda
+        packages:
+          - libcuvs==25.12.*,>=0.0.0a0
```
Should this be 26.02?

alexbarghi-nv replied:
I already have the change done locally

alexbarghi-nv replied:
Ok pushed the fix, sorry about that.

rlratzel committed:
Requires NX 3.5 for edge BC tests (#5355)

* Requires that edge BC tests use NX 3.5 for comparisons, otherwise those tests are XFAIL'd.
  * Changes to NX 3.6 caused some comparisons in tests to fail when NX 3.6 is used as the reference. We're essentially skipping for now but will look into those changes (likely [here](https://github.com/networkx/networkx/pull/8256))
* This also updates the `requires_nx` marker to take up to 2 args for min and max required version, and a flag for skip vs. xfail.

Authors:
  - Rick Ratzel (https://github.com/rlratzel)
  - Brad Rees (https://github.com/BradReesWork)

Approvers:
  - Brad Rees (https://github.com/BradReesWork)
  - Don Acosta (https://github.com/acostadon)

URL: https://github.com/rapidsai/cugraph/pull/5355

alexbarghi-nv committed:
Merge branch 'release/25.12' of https://github.com/rapidsai/cugraph into main-merge-release/25.12
